src/kernel/kernel.c

```diff
@@ -359,8 +389,10 @@ kthread_t* kernel_init(void)
 	th_main.name = "main";
 	th_main.stack_size = 0x10000; // see STACK_SIZE in system.c
 	th_main.flags = TH_FLAG_DETACHED; // main thread cannot be joined
+	th_main.tp_value = th_main_tp;
```

Comment:
th_main_tp seems to be a constant. Any reason to define it as a variable, rather than just using `__tls_base+TP_OFFSET` here?

Reply:
That also works, so done.